src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package duke;
+
+import java.text.ParseException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+/**
+ * Represents a task that has a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDateTime deadline;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM d yyyy, hh:mm a");
+    private static final DateTimeFormatter SAVE_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```

Comment:
I think it's good practice to have static objects on top.
```suggestion
    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("MMM d yyyy, hh:mm a");
    private static final DateTimeFormatter SAVE_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
    private LocalDateTime deadline;
```